### ISSUE TITLE
Rewrite entire app in Kotlin

### ISSUE THREAD

**Issue by y9san9** [Sketchware-Pro/Sketchware-Pro#144]

**Comment by JavkhlanK** [comment]
why though, the app works fine lol

**Comment by y9san9** [comment]
Everybody knows Kotlin is the best language, so just wonder Sketchware Pro in Kotlin

**Comment by y9san9** [comment]
You can only have monoids in category of endofunctors in Kotlin, not Java

**Comment by JavkhlanK** [comment]
alright sounds cool, but does Sketchware Pro need it? I don't think so.

**Comment by dmitrijkotov634** [comment]
i want to see the source code of Sketchware on kotlin. Kotlin is the best programming language.

**Comment by y9neon** [comment]
Totally agree with @dmitrijkotov634 and @y9san9

**Comment by crinnyx** [comment]
Definetely rewrite it.

**Comment by dvdplayer6767** [comment]
Yes, you have to do it. 100%

**Comment by ImGonnaSpamStarLol** [comment]
imagine teamwork with 7 people just for a dumb idea

**Comment by JavkhlanK** [comment]
rewrite for what tho? there aren't any benefits of it right now, and we can't re-architecture the whole app

**Comment by Iyxan23** [comment]
I'd say it's an interesting idea, kotlin is great for writing less codes and good for focusing on what you want to do, It can make the source code more organized and readable. Although, It would be a very big consideration as kotlin does has it's ups and downs impact to this project.

And as JavkhlanK said, "we can't re-architecture the whole app", It would be better for you guys to create a new sketchware from scratch in kotlin rather than asking a very big long-existed java-dependant project to convert it's entire source to kotlin.

In short: we're not ready yet.

**Comment by DantaNotEpic** [comment]
<a:sad:830527448005345280> 🙏

**Comment by swgames-dev** [comment]
I love kotlin!!!

**Comment by khaled-0** [comment]
no, java is love

**Comment by yallxe** [comment]
Totally agree.
